pipes/connection_config.go

```diff
@@ -55,27 +54,27 @@ func connect(_ context.Context, d *plugin.QueryData) (*openapiclient.APIClient,
 		host = os.Getenv("PIPES_HOST")
 	}
 	// host value present in the config takes precedence over environment variable
-	if pipesConfig.Host != nil {
-		host = *pipesConfig.Host
+	if config.Host != nil {
+		host = *config.Host
 	}
 
-	if host != "" && !strings.Contains(host, "cloud.steampipe.io") && !strings.Contains(host, "pipes.turbot.com") {
-		parsedURL, parseErr := url.Parse(host)
-		if parseErr != nil {
-			return nil, fmt.Errorf(`invalid host: %v`, parseErr)
-		}
-		if parsedURL.Host == "" {
-			return nil, errors.New(`missing protocol or host`)
-		}
+	parsedURL, parseErr := url.Parse(host)
```

Comment:
@sidr0cker @MichaelBurgess should we add a condition to check if the `host` config parameter has been set or not?

IMO, since `host` is an optional config argument, the  `parsedURL, parseErr := url.Parse(host)` might return an error saying `invalid host`, which is opposite to what the documentation says about `host` being an optional parameter.

I think we should set a default value if `host` is not set.

Reply:
@misraved if the `host` is not set, we will default it to `pipes.turbot.com`.

The reason for making this change is to support custom tenants. The SDK always assumes the host to be `pipes.turbot.com`. Hence, if the host set for the plugin is anything other than production primary tenant, we need to parse the host and set the appropriate host the be used by the SDK.
So we need to parse the host every time, but do nothing if the host is `pipes.turbot.com`.

I'll make the change now.